File: modules/common-model/src/main/java/com/shterneregen/securelan/common/model/ChatEnvelope.java
package com.shterneregen.securelan.common.model;

import java.util.Objects;

public record ChatEnvelope(
        ChatMessage message,
        DeliveryStatus status
) {
    public ChatEnvelope {
        Objects.requireNonNull(message, "message must not be null");
        Objects.requireNonNull(status, "status must not be null");
    }
}

File: modules/common-model/src/main/kotlin/com/shterneregen/securelan/common/model/ChatEnvelope.kt
package com.shterneregen.securelan.common.model

import java.util.Objects

@JvmRecord
data class ChatEnvelope(
    val message: ChatMessage,
    val status: DeliveryStatus,
) {
    init {
        Objects.requireNonNull(message, "message must not be null")
        Objects.requireNonNull(status, "status must not be null")
    }
}
